activitypub/database/database.go

```diff
@@ -18,22 +24,48 @@ import (
 )
 
 var (
-	regexpID     = regexp.MustCompile("^https://(.+)/([^/]+)$")
-	regexpOutbox = regexp.MustCompile("^https://(.+)/outbox$")
-	regexpInbox  = regexp.MustCompile("^https://(.+)/inbox$")
+	regexpID        = regexp.MustCompile("/user/([^\\/]+)$")
+	regexpOutbox    = regexp.MustCompile("/user/([^\\/]+)/outbox$")
+	regexpInbox     = regexp.MustCompile("/user/([^\\/]+)/inbox$")
+	regexpRead      = regexp.MustCompile("/user/([^\\/]+)/read/(.*)")
+	regexpFollowers = regexp.MustCompile("/user/([^\\/]+)/followers")
+)
+
+const (
+	// ResultsPerPage is how many results to return in a response
+	ResultsPerPage = 10
+
+	// StartPage is the number of pages to skip before returning results
+	StartPage = 0
 )
 
 // A Database is a connection to a database. It uses the gorm connection, so that we can still use the models.
 type Database struct {
-	DB    *gorm.DB
-	locks map[*url.URL]*sync.Mutex
+	baseURL    string
+	DB         *gorm.DB // TODO needed?
```

Comment:
I would prefer that anything database related remains in the infrastructure package, so we should rarely if ever need gorm.DB anywhere else besides main and, for ease of instantiating, the New method of things needing repositories.

Reply:
Changing